packages/api-reference/src/v2/helpers/map-configuration.ts

```diff
@@ -0,0 +1,56 @@
+import type { ApiReferenceConfigurationWithSources } from '@scalar/types'
+import type { DocumentConfiguration } from '@scalar/workspace-store/client'
+
+/**
+ * Maps a partial ApiReferenceConfigurationWithSources object to a DocumentConfiguration
+ * for use in the API Reference. This function transforms configuration options
+ * into the expected structure for the 'x-scalar-reference-config' property.
+ *
+ * @param config - Partial configuration object for the API Reference
+ * @returns DocumentConfiguration object with mapped settings
+ */
+export const mapConfiguration = (config: Partial<ApiReferenceConfigurationWithSources>) => {
```

Comment:
you probably want `ApiReferenceConfiguration` for a single config
OR
`AnyApiReferenceConfiguration` to take in all the configs